### PR TITLE
Feature/320 add matrix test

### LINKED ISSUE #320: improve code coverage in MatrixSupport
### Task description

Sonar shows many uncovered lines in MatrixSupport. Also some minor issues are found by Sonar.
We have unit tests for that part, but apparently misses it some corner cases.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by epsig**

### CODE REVIEW

File: sources/Deltares.Probabilistic/Math/MatrixSupport.cpp [hunk 1]
```diff
@@ -26,8 +26,8 @@
 
 namespace Deltares::Numeric
 {
-    const double SMALL_NUM = 1e-10;
-    const int MAX_INT = 1215752192;
+    constexpr double SMALL_NUM = 1e-10;
+    constexpr int MAX_INT = 1215752192;
```
Comment (robdeltares) [inline sources/Deltares.Probabilistic/Math/MatrixSupport.cpp]: This value is not the maximum integer value in C++, please rename or assign value from std:;limits

**Review by robdeltares — CHANGES_REQUESTED** [review]

**Review by robdeltares — APPROVED** [review]
